### ISSUE TITLE
Get all the Table data when a button is clicked

### ISSUE THREAD

**Issue by DVGY** [silevis/reactgrid#168]
Is there any way to get all the table data, I am sure ReactGrid is storing all these cell data in some Data structure. Is there any way I can get List of Row Object (Which contains all cell data) or table object 

For example here [code sandbox](https://codesandbox.io/s/reactgrid-calculated-cell-on-change-rsmzd8?file=/src/index.tsx)

Is there anyway way I can get all the table data ? maybe a hook? or some func?
<img width="389" alt="image" src="https://github.com/silevis/reactgrid/assets/46158826/30d4bd0f-99be-4a25-b851-cf930368a0cc">

**Comment by DVGY** [comment]
Update: I found that using ref I can access the table data. Is there any other recommended way ?

**Comment by alonshmiel** [comment]
I'm not sure I understand your question but ReactGrid gets rows that contain all the rows of the table.

When rowId is 'header', it means that this row is the header.
Every row contains its cells data.

**Comment by DVGY** [comment]
> cells data

Okay let me try to explain again

```js
function App() {
  const [additionRows] = React.useState(getAdditionRows());
  const [rows, setRows] = React.useState(getRows(additionRows));
  const columns = getColumns();
  const tableRef = React.useRef();

  console.log(tableRef);

  const handleChanges = (changes: CellChange<TextCell>[]) => {
    setRows((prevRows) => applyChangesToRows(changes, prevRows));
  };
  return (
    <>
      <ReactGrid
        ref={tableRef}
        rows={rows}
        columns={columns}
        onCellsChanged={handleChanges}
      />
    </>
  );
}

```

Value of console log is 

```
current: n
props: Object
rows: Array(2)
0: Object
1: Object
columns: Array(3)
onCellsChanged: ƒ handleChanges() {}
context: Object
refs: Object
updater: Object
updateState: ƒ () {}
stateUpdater: ƒ () {}
pointerEventsController: t
eventHandlers: Pe
cellMatrixBuilder: e
state: Object
_reactInternals: FiberNode
_reactInternalInstance: Object
<constructor>: "n"
componentDidUpdate: ƒ () {}
componentDidMount: ƒ () {}
componentWillUnmount: ƒ () {}
render: ƒ () {}
isReactComponent: Object
setState: ƒ () {}

```

I need this 
<img width="148" alt="image" src="https://github.com/silevis/reactgrid/assets/46158826/8bf9961c-efa3-4bd1-a994-180059546ee6">

Is there any other way to get these values ?

**Comment by alonshmiel** [comment]
Sorry but doesn't it work?

console.log(rows);

**Comment by DVGY** [comment]
Yaa this also works. I am closing this issue Thanks for your help
